### PR TITLE
feat: Sumologicextension otlp traces support

### PR DESCRIPTION
**Pull request by pmatyjasek-sumo**
feat: Sumologicextension otlp traces support

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

### CODE REVIEW

File: pkg/exporter/sumologicexporter/factory.go [hunk 1]
```diff
@@ -87,3 +88,16 @@ func createMetricsExporter(
...
 }
+
+func createTracesExporter(
+	_ context.Context,
+	params component.ExporterCreateParams,
+	cfg config.Exporter,
+) (component.TracesExporter, error) {
+	exp, err := newTracesExporter(cfg.(*Config), params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the metrics exporter: %w", err)
```
Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/factory.go:99]:
```suggestion
		return nil, fmt.Errorf("failed to create the traces exporter: %w", err)
```

**Comment by sumo-drosiek** [comment]
I would add comments for function, but generally good to go

**Comment by sumo-drosiek** [comment]
@pmatyjasek-sumo  And please update `README.md`

**Comment by pmatyjasek-sumo** [comment]
@sumo-drosiek sure, I'll update comments and README. Thanks for review :)

File: pkg/exporter/sumologicexporter/sender.go [hunk 2]
```diff
@@ -522,6 +547,16 @@ func addMetricsHeaders(req *http.Request, mf MetricFormatType) error {
 	return nil
 }
 
+func addTracesHeaders(req *http.Request, tf TraceFormatType) error {
+	switch tf {
+	case OTLPTraceFormat:
+		req.Header.Add(headerContentType, contentTypeOTLP)
+	default:
+		return fmt.Errorf("unsupported tracess format: %s", tf)
```
Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/sender.go:555]:
```suggestion
		return fmt.Errorf("unsupported traces format: %s", tf)
```

**Comment by sumo-drosiek** [comment]
Sorry for additional request, but please add config validation like in logs/metrics: https://github.com/SumoLogic/sumologic-otel-collector/blob/c1fd99669cd8579d435840f3801d99c20d8a946e/pkg/exporter/sumologicexporter/exporter.go#L60-L67

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**